packages/web3torrent/src/utils/checkTorrentInTracker.ts

```diff
@@ -1,33 +1,33 @@
 import {Client} from 'bittorrent-tracker';
 import {defaultTrackerOpts} from '../constants';
 import {TorrentTestResult} from '../library/types';
+import debug from 'debug';
+const log = debug('web3torrent:tracker-check');
 
 export async function checkTorrentInTracker(infoHash: string) {
+  log(`Scrapping tracker for torrent ${infoHash}`);
   const trackerClient: Client = new Client({
     ...defaultTrackerOpts,
     infoHash: [infoHash]
   });
-  let completePeers;
-  const gotAWire: Promise<boolean> = new Promise(resolve => {
-    const updateIfSeederFound = data => {
-      completePeers = data.complete;
-    };
-    trackerClient.once('peer', function() {
-      resolve(true);
-    });
-    trackerClient.once('update', updateIfSeederFound);
-    trackerClient.start();
+  const trackerScrape: Promise<number> = new Promise(resolve => {
+    trackerClient.once('scrape', data => resolve(data.complete));
+    setTimeout(() => trackerClient.scrape(), 2500); // waits ~2 seconds as that's the tracker update tick
   });
   const timer: Promise<undefined> = new Promise((resolve, _) => setTimeout(resolve, 5000));
-  const raceResult = await Promise.race([gotAWire, timer]);
-  trackerClient.stop();
-  trackerClient.destroy();
-  if (raceResult) {
-    return TorrentTestResult.SEEDERS_FOUND; // Could connect to a peer
-  }
-  if (Number.isInteger(completePeers)) {
-    // was able to connect to the tracker, but couldn't connect to a peer
-    return TorrentTestResult.NO_SEEDERS_FOUND;
+
+  const raceResult = await Promise.race([trackerScrape, timer]);
```

Comment:
Is the race still needed now that scrape is called after 2.5 seconds?

Reply:
umm... haven't thought about it. I'll test it out tomorrow, and if it's not, I'll make a new PR.